### PR TITLE
docs: change global configuration order wording

### PR DESCRIPTION
**Pull request by ricardovanlaarhoven**
## Description
The order of global-configuration was wrong. When i read it, it seemed odd to me since you'd think that the component specific would overwrite the global section. After testing it i saw that it indeed had a logical order and that's why the update to the docs is needed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by KaelWD** [comment]
This was already correct. Buttons will be green with the following:

```
defaults: {
  global: { color: 'red' },
  VBtn: { color: 'green' }
}
```

**Comment by ricardovanlaarhoven** [comment]
> This was already correct. Buttons will be green with the following:
> 
> ```
> defaults: {
>   global: { color: 'red' },
>   VBtn: { color: 'green' }
> }
> ```


The old doc order says:

Value defined in component specific section of defaults configuration object
Value defined in global section of defaults configuration object
Value defined in the component definition

Which i read as:
it first uses the components default configuration
Which is overwritten by the global configuration
Which is overwritten by the component definition

But it first checks the global configuration and that is overwritten by the component's default configuration. Not the other way around.

Or am I missing something?

**Comment by KaelWD** [comment]
Yeah it's more like  `prop ?? defaults.component ?? defaults.global ?? component.props.default`

**Comment by ricardovanlaarhoven** [comment]
> Yeah it's more like `prop ?? defaults.component ?? defaults.global ?? component.props.default`

But what's described in the documentation is
`prop ?? default.global ?? defaults.component`
And i changed it to
`prop ?? defaults.component ?? default.global`

**Comment by KaelWD** [comment]
You might be reading it backwards? It isn't 1 overrides 2 overrides 3, it's 1 otherwise 2 otherwise 3
If you think you can clarify the wording a bit go ahead, but the order is already correct

**Comment by jacekkarczmarczyk** [comment]
IMHO current order is ok - first check for component's specific defaults, if not provided the globals, if not provided the default value from prop definition

BTW - shouldn't it include also default values provided by `VDefaultsProvider`?

**Comment by KaelWD** [comment]
Yeah that's kinda complicated because VDefaultsProvider merges then replaces the global configuration, so it'd be
`prop ?? defaultsProvider.component ?? defaultsProvider.global ?? globalDefaults.component ?? globalDefaults.global ?? component.props.default`

**Comment by ricardovanlaarhoven** [comment]
Ah now I know why I was confused. I thought you meant the creation of the component `<VBtn color="green"/>` with " component definition". But it's the `default` inside a `prop` inside the `VBtn`

File: packages/docs/src/pages/en/features/global-configuration.md [hunk 1]
```diff
@@ -39,8 +39,8 @@ export default createVuetify({
 
 When creating and mounting a component, Vuetify uses the following priority in determining which default prop value to use:
 
-1. Value defined in component specific section of `defaults` configuration object
-2. Value defined in global section of `defaults` configuration object
+1. Value defined in global section of `defaults` configuration object
+2. Value defined in component specific section of `defaults` configuration object
 3. Value defined in the component definition
```
Comment (ricardovanlaarhoven) [inline packages/docs/src/pages/en/features/global-configuration.md:44]:
```suggestion
1. Value set as prop value to the component itself
2. Value defined in component specific section of defaults configuration object
3. Value defined in global section of defaults configuration object
4. Value defined in the component definition by Vuetify
```

**Comment by jacekkarczmarczyk** [comment]
> which default prop value to use:

So either "value set as a prop ..." shouldn't be on the list or the bit above should be changed

File: packages/docs/src/pages/en/features/global-configuration.md [hunk 2]
```diff
@@ -39,8 +39,9 @@ export default createVuetify({
 
 When creating and mounting a component, Vuetify uses the following priority in determining which default prop value to use:
```
Comment (ricardovanlaarhoven) [inline packages/docs/src/pages/en/features/global-configuration.md:40]:
```suggestion
When creating and mounting a component, Vuetify uses the following priority in determining which prop value to use:
```

File: packages/docs/src/pages/en/features/global-configuration.md [hunk 3]
```diff
@@ -39,8 +39,9 @@ export default createVuetify({
...
 
-1. Value defined in component specific section of `defaults` configuration object
-2. Value defined in global section of `defaults` configuration object
-3. Value defined in the component definition
+1. Value set as prop value to the component itself
+2. Value defined in component specific section of defaults configuration object
+3. Value defined in global section of defaults configuration object
+4. Value defined in the component definition by Vuetify
```
Comment (ricardovanlaarhoven) [inline packages/docs/src/pages/en/features/global-configuration.md:45]:
```suggestion
1. Value set as prop value to the component itself
2. Value defined in component specific section of defaults configuration object
3. Value defined in global section of defaults configuration object
4. Value defined in the prop definition of the Vueitfy component itself.
```

**Comment by ricardovanlaarhoven** [comment]
If you guys think it's clear enough you can close this PR. Me and some coworkers just didn't get it and i hope this prevents some questions about this.

**Pull request merged**